src/main/kotlin/org/jitsi/nlj/rtcp/RtcpEventNotifier.kt

```diff
@@ -26,17 +26,21 @@ import java.time.Instant
  */
 // TODO(brian): maybe post the notifications to another pool, so we don't hold up the caller?
 class RtcpEventNotifier {
-    private val rtcpListeners: MutableList<RtcpListener> = CopyOnWriteArrayList<RtcpListener>()
+    private val rtcpListeners: MutableList<Pair<RtcpListener, Boolean>> = CopyOnWriteArrayList()
 
-    fun addRtcpEventListener(listener: RtcpListener) {
-        rtcpListeners.add(listener)
+    fun addRtcpEventListener(listener: RtcpListener, external: Boolean = false) {
+        rtcpListeners.add(Pair(listener, external))
     }
 
-    fun notifyRtcpReceived(packet: RtcpPacket, receivedTime: Instant?) {
-        rtcpListeners.forEach { it.rtcpPacketReceived(packet, receivedTime) }
+    fun notifyRtcpReceived(packet: RtcpPacket, receivedTime: Instant?, external: Boolean = false) {
+        rtcpListeners.forEach { (listener, listenerIsExternal) ->
+            if (!external || !listenerIsExternal) listener.rtcpPacketReceived(packet, receivedTime)
```

Comment:
Can you please document the semantics of "external"? I fail to understand this intuitively -- external packets are only forwarded to non-external listeners?